products/green_fast_track/macros/clip_to_geom.sql

```diff
@@ -18,8 +18,13 @@ from "left" table and the resulting geom column named "geom"
                 {{ left }}.{{ column }},
             {% endfor %}
         {% endif %}
-        ST_INTERSECTION({{ left }}.{{ left_by }}, {{ right }}.{{ right_by }}) AS geom
+        CASE
+            WHEN ST_COVEREDBY({{ left }}.{{ left_by }}, {{ right }}.{{ right_by }})
+                THEN {{ left }}.{{ left_by }}
+            ELSE ST_INTERSECTION({{ left }}.{{ left_by }}, {{ right }}.{{ right_by }})
+        END
+         AS geom
     FROM {{ left }}
-    INNER JOIN {{ right }} ON ST_INTERSECTS({{ left }}.{{ left_by }}, {{ right }}.{{ right_by }})
+    INNER JOIN {{ right }} ON ST_RELATE({{ left }}.{{ left_by }}, {{ right }}.{{ right_by }}, 'T********')
```

Comment:
It seems we shouldn't be using st_intersects in general if we are looking for overlapping polygons. If they share a line border, that line is returns as the intersection

https://gis.stackexchange.com/questions/273993/st-relate-intersect-ignore-shared-boundary

Reply:
The perf part of this commit is the upper part though - only calculate intersection if need by, because st_coveredby is much faster than st_intersection

See https://postgis.net/documentation/tips/tip_intersection_faster/

Reply:
Good to know! I wonder if (outside of this macro), 
```
            WHEN ST_COVEREDBY(...)
                THEN orig.geom
            ELSE ST_INTERSECTION(...)
```
warrants a macro of it's own

Reply:
Or just a postgres function so we don't need to use jinja. given that this is in the official docs, I'm surprised they don't just have an out of the box function that does this